src/components/LayerTree/LayerTree.js

```diff
@@ -151,7 +158,9 @@ class LayerTree extends Component {
     const { layerService } = this.props;
     if (layerService) {
       this.updateLayers(layerService);
-      layerService.on('change:visible', () => this.updateLayers(layerService));
+      this.onChangeRef = layerService.on('change:visible', () =>
```

Comment:
With this method you have to `un` before the  `on`  because the internale `un` will not work since you create another callback.